app/src/main/java/com/example/moviedb/ui/adapter/MovieAdapter.kt

```diff
@@ -11,7 +11,7 @@ import com.example.moviedb.data.model.Movie
 import com.example.moviedb.databinding.ItemMovieBinding
 import com.example.moviedb.utils.ImageUtils
 
-class MovieAdapter : Adapter<ViewHolder>() {
+class MovieAdapter(val listener: OnItemClickListener) : Adapter<ViewHolder>() {
```

Comment:
dùng high order function đi anh eii 
```suggestion
class MovieAdapter (
    private val onItemClickListener: (Movie) -> Unit 
) : Adapter<ViewHolder>() {
```